react_main/src/pages/Game/Game.jsx

```diff
@@ -1544,10 +1544,45 @@ export function SideMenu(props) {
     );
 }
 
+function RoleMarkerToggle(props) {
+    const roleMarkerRef = useRef();
+    const popover = useContext(PopoverContext);
+    const game = useContext(GameContext);
+    const { toggleRolePrediction } = game;
+    const playerId = props.playerId;
+
+    function onRoleMarkerClick() {
+		if (props.onClick)
+			props.onClick();        
+        
+        popover.onClick(
+            `/setup/${game.setup.id}`,
+            "rolePrediction",
+            roleMarkerRef.current,
+            "Mark Role as",
+            data => {
+                data.roles = JSON.parse(data.roles)[0];
+                console.log("2")
+                data.toggleRolePrediction = toggleRolePrediction(playerId);
+            }
+        )
+    }
+
+    return (
+        <div className="role-marker" 
+            onClick={onRoleMarkerClick}
+            ref={roleMarkerRef}>
+                
+            <i className="fas fa-map-pin"></i>

```

Comment:
i'm ok to change to another icon, i don't particularly like this but i'm not sure what others are good